src/Common/src/Internal/Cryptography/UniversalCryptoEncryptor.cs

```diff
@@ -50,6 +51,28 @@ private byte[] PadBlock(byte[] block, int offset, int count)
                     Buffer.BlockCopy(block, offset, result, 0, result.Length);
                     break;
 
+                // ANSI padding fills the blocks with zeros and adds the total number of padding bytes as
+                // the last pad byte, adding an extra block if the last block is complete.
+                //
+                // x 00 00 00 00 00 00 07
+                case PaddingMode.ANSIX923:
+                    result = new byte[count + padBytes];
+                    Buffer.BlockCopy(block, 0, result, 0, count);
```

Comment:
`Buffer.BlockCopy(block, 0, result, 0, count);` 
should be
`Buffer.BlockCopy(block, offset, result, 0, count);`

and same for the other one.